### PR TITLE
While loading each module try catch every exception, and log it.

### LINKED ISSUE #165: Application crash while looking for module in JDBC library
### Description
When Discord4J tries to load JDBC class (in my case - Xerial's JDBC), Class.forName() causes triggered actions inside this class BEFORE declaration of another required classes (? not sure what's actually going on inside JDBC class), so we have exception like this: http://pastebin.com/Dnr7a3p9
In fact, this can happen with ANY another class in another libraries that has static { } blocks under certain circumstances. So, basicly, the simplest and the best solution avaible is to remove static module loading from Discord4J, Class.forName() is really bad. This occurs at sx.blah.discord.modules.ModuleLoader.loadExternalModules(ModuleLoader.java:223), as you can see from exception higher.

### Steps to Reproduce
1. Create Discord4J module
2. Add class that uses static {} blocks in way it uses libraries like xerial's JDBC (or add library like it)
3. Profit, you're crashed forever!

...

**Expected behavior:** module successfully loaded and everything works well

**Actual behavior:** Class.forName() used to load class breaks everything

**Stacktrace (if applicable):** http://pastebin.com/Dnr7a3p9

**Version affected:** 2.6.1

P.S. Sorry for my english if it is bad :(

### PR DESCRIPTION
**Pull request by ArsenArsen**
### Prerequisites
* [x] I have read and understood the [contribution guidelines](CONTRIBUTING.md)
* [x] This pull request follows the code style of the project
* [x] I have tested this feature thoroughly
  * [Proof of testing](http://imgur.com/a/8p8KF)

**Issues Fixed:** One module's load failure would result in a crash of the entire bot.

### Changes Proposed in this Pull Request
* While loading each module try catch every exception, and log it.

Fixes #165

### CODE REVIEW

**Comment by theIglooo** [comment]
Wrapping everything in a try/catch that catches Exception certainly is a hacky "fix", and you aren't supposed to handle errors.
This also doesn't fix whatever is causing his module to fail assuming it's on our end, it just tells you in a nicer way that it failed.

**Comment by ArsenArsen** [comment]
#156 is not Discord4J's end, this PR is aimed to only handles errors in module initialization better. Modules are loading 3rd parties, and not all 3rd party code is trust-worthy. Beforehand an error in user's code would crash the bot, and that's not good. This PR, as I said, is aimed to fix that issue. #165 is just an example of that issue.

File: src/main/java/sx/blah/discord/modules/ModuleLoader.java [hunk 1]
```diff
@@ -236,7 +242,7 @@ public static synchronized void loadExternalModules(File file) { // A bit hacky,
 							addModuleClass(classInstance);
 					}
 				}
-			} catch (NoSuchMethodException | IllegalAccessException | InvocationTargetException | IOException | ClassNotFoundException e) {
+			} catch (NoClassDefFoundError | NoSuchMethodException | IllegalAccessException | InvocationTargetException | IOException | ClassNotFoundException e) {
```
Comment (ArsenArsen) [inline src/main/java/sx/blah/discord/modules/ModuleLoader.java]: No. NoClassDefinitionError is an error hence error in its name. Errors are not exceptions.

Comment (GrandPanda) [inline src/main/java/sx/blah/discord/modules/ModuleLoader.java]: Correct, errors should almost never be caught.

**Comment by GrandPanda** [comment]
Seeing as the premise for this PR is no longer valid, I'm going to close it.

**Pull request closed without merging**